lsqecc/pauli_rotations/circuit.py

```diff
@@ -154,7 +155,8 @@ def remove_y_operators_from_circuit(self, start_index: int = 0) -> None:
                         while right_block_index + 1 < len(self.ops):
                             self.commute_pi_over_four_rotation(right_block_index)
                             right_block_index += 1
-                        self.ops.pop()
+                        if self.circuit_has_measurements():
```

Comment:
Since `self.circuit_has_measurements()` is a loop itself, it is better to call it outside of the loop at the top.